assets/js/components/BaseNews.vue

```diff
@@ -0,0 +1,144 @@
+    list-style-type: none;
+  }
+
+  ul {
+    padding-left: 0;
+  }
+
+  small {
+    color: #898989;
+  }
+
+  img {
+    float: left;
+    width: 40%;
+    margin-right: 1rem;
+  }
+
+  .title {
+    font-size: 1.3rem;
+  }
+
+}
+</style>
+
+
+<script>
+import * as d3 from 'd3';
+import _ from 'lodash';
+
+
+import Component from './Component';
+
+
+export default Component.extend({
+  computed: {
+    data() {
+      if (!this.hasData) return []
+
+      const dataset = this.filtered;
+      const unique = {};
+
+      // use dict to remove duplicates
+      for (const d of dataset) {
+        for (const news of d.news){
+          unique[news.link] = news;
+        }
+      }
+
+      const out = this.getSortedNewsForNUTS(unique) || [];
+
+      return out;
+    },
+  },
+
+  methods: {
+    /**
+     * @param {Object} all_news
+     * @returns {Object[]} news - news for specific region (nuts), ordered by relevance
+     */
+    getSortedNewsForNUTS(all_news) {
+      const deep_search = true;
+      const news_for_nuts = this.getNewsForRegion(all_news, this.localfilters.region || "", deep_search);
+
+      // sometimes this.localfilters.region is null (for filtering by country or removing filters)
+      if(this.localfilters.region) {
+        // sort by relevance and created date
+        _.orderBy(news_for_nuts, ["nuts", "created"], ["desc", "desc"])
+      } else {
+        // sort by created date
+        news_for_nuts.sort((a,b) => d3.descending(a.created,b.created));
+      }
+
+      return news_for_nuts;
+    },
+    /**
+     * recursive function that returns news for a region
+     * @param {Object} all_news
+     * @param {string|null} region - will find news for region, for null, it will return all news
+     * @param {boolean} deep_search - find news for included areas, only the first time, to avoid duplicating news
+     * @returns {Object[]} filtered_news - news for specific region (nuts), ordered by relevance, looks for parent region news if
+     * news for region is less than 3 ex: for RO31, return RO31, RO311, RO312, if results are less than 3, return RO3, or RO, 
+     * it will not return from neighbor regions like RO32
+     */
+    getNewsForRegion(all_news, region, deep_search) {
+      let filtered_news = [];
+
+      for (const link in all_news) {
+        const nuts = all_news[link].nuts;
+        if(deep_search) {
+          // if region is "RO12" it will include nuts like "RO12" and "RO121"
+          if(nuts.substr(0, region.length) === region || !region) {
+            filtered_news.push(all_news[link]);
+          }
+        } else {
+          // if region is "RO12" it will only take nuts = "RO12"
+          if(nuts === region || !region) {
+            filtered_news.push(all_news[link]);
+          }
+        }
+      }
+      deep_search = false;
+      // search for parent news if region news are less than 3, stop ar country code ex: "RO"
+      if(filtered_news.length <= 3 && region.length >= 2) {
```

Comment:
region is either null and this will go boom (right?) or at least a beneficiary state, which already has .length = 2

Reply:
when a beneficiary state is selected, or a region like RO3, region comes as null, which is changed into "" by getSortedNewsForNUTS().